app/assets/javascripts/orders.js

```diff
@@ -46,10 +46,22 @@ ready = function() {
     increment_product($(this).data("product"))
   })
 
+  $("#barcode_id").keyup(function() {
+    barcode = $("#barcode_id").val();
+    if (! (/^\d*$/.test(barcode))) {
```

Comment:
Are there barcodes with spaces and/or dashes?

Reply:
No, or at least not that we use